resource-aggregate/cqrs/events/resourceStateSnapshotTaken.go

```diff
@@ -114,28 +89,26 @@ func (rs *ResourceStateSnapshotTaken) ValidateSequence(eventMetadata *pb.EventMe
 	return false
 }
 
-func (rs *ResourceStateSnapshotTaken) HandleEventResourceChanged(ctx context.Context, contentChanged ResourceChanged) (bool, error) {
-	if rs.ValidateSequence(contentChanged.EventMetadata) {
+func (rs *ResourceStateSnapshotTaken) HandleEventResourceChanged(ctx context.Context, contentChanged *ResourceChanged) (bool, error) {
+	if rs.ValidateSequence(contentChanged.GetEventMetadata()) {
 		rs.LatestResourceChange = &contentChanged.ResourceChanged
 		return true, nil
 	}
 	return false, nil
 }
 
-func (rs *ResourceStateSnapshotTaken) HandleEventResourceDeleted(ctx context.Context, deleted ResourceDeleted) error {
+func (rs *ResourceStateSnapshotTaken) HandleEventResourceDeleted(ctx context.Context, deleted *ResourceDeleted) error {
 	return nil
 }
 
-func (rs *ResourceStateSnapshotTaken) HandleEventResourceStateSnapshotTaken(ctx context.Context, s ResourceStateSnapshotTaken) error {
-	if s.PendingRequestsCount != 0 {
+func (rs *ResourceStateSnapshotTaken) HandleEventResourceStateSnapshotTaken(ctx context.Context, s *ResourceStateSnapshotTaken) error {
+	if s.GetPendingRequestsCount() != 0 {
 		return status.Errorf(codes.FailedPrecondition, "invalid pending requests")
 	}
-	rs.Id = s.Resource.Id
-	rs.Resource = s.Resource
-	rs.LatestResourceChange = s.LatestResourceChange
-	rs.TimeToLive = s.TimeToLive
-	rs.IsPublished = s.IsPublished
-	rs.EventMetadata = s.EventMetadata
+	rs.Id = s.GetResource().GetId()
+	rs.Resource = s.GetResource()
```

Comment:
What is stored at `rs.Resource` ? I think it was moved to resource link so it is empty. It can be used for store "href, id, deviceid".